### PR TITLE
Fix credential creation flow for msteams app

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

After some bug/issues on prod I took a look of what could be the error and found out some scenarios that wasn't handling well the credential update/creation for this app.

- Avoid duplicated credentials type office365_key
- Fix scope when renewing token

Fixes # (issue)

**Environment**: Staging(main branch) / Production

## Type of change

Loom: https://www.loom.com/share/cc6b2a5d42b44301bb2a37786e915b5b

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- Install MSteam app
- Add Office365video as Location for one of your event types
- Book with said event type many times with some time difference between them, so your current credential expires and gets renewed.
- You should not get error when creating an event after some time has passed.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by joeauyeung — APPROVED** [review]
LGTM

**Comment by PeerRich** [comment]
@alannnc can you do a self-review and loom video?

File: packages/app-store/office365video/api/callback.ts [hunk 1]
```diff
@@ -65,14 +65,39 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
-  await prisma.credential.create({
-    data: {
-      type: "office365_video",
-      key: responseBody,
-      userId: req.session?.user.id,
-      appId: "msteams",
-    },
-  });
+  const userId = req.session?.user.id;
+  if (userId) {
```
Comment (leog) [inline packages/app-store/office365video/api/callback.ts:69]: Might be useful to apply the pattern we conveyed as a best practice: [Prefer early returns](https://app.gitbook.com/o/6snd8PyPYMhg0wUw6CeQ/s/VXRprBTuMlihk37NQgUU/engineering/best-practices/prefer-early-returns)

**Review by leog — COMMENTED** [review]
LGTM, suggested a change to align to best practices

File: packages/app-store/office365video/api/callback.ts (hunk 1 above)
Comment (zomars) [reply packages/app-store/office365video/api/callback.ts:69]: Agreed

Comment (alannnc) [reply packages/app-store/office365video/api/callback.ts:69]: solved

**Review by zomars — APPROVED** [review]

**Pull request merged**